File: typescript/src/iconsumer_api.ts
import {APIs} from './apis/index';

/**
 * The primary access point for the ConsumerAPI.
 * This service contains all functions that the ConsumerAPI employs to
 * communicate with the ProcessEngine.
 */
export interface IConsumerApi
  extends APIs.IBoundaryEventConsumerApi,
  APIs.ICallActivityConsumerApi,
  APIs.IEmptyActivityConsumerApi,
  APIs.IEventConsumerApi,
  APIs.IIntermediateEventConsumerApi,
  APIs.IManualTaskConsumerApi,
  APIs.INotificationConsumerApi,
  APIs.IProcessModelConsumerApi,
  APIs.IUserTaskConsumerApi {}

File: dotnet/src/IConsumerAPI.cs
namespace ProcessEngine.ConsumerAPI.Contracts
{
    using ProcessEngine.ConsumerAPI.Contracts.APIs;

    /// <summary>
    /// API for starting and managing BMPN processes.
    /// </summary>
    public interface IConsumerAPI :
        IEmptyActivityConsumerApi,
        IEventConsumerApi,
        IManualTaskConsumerApi,
        IProcessModelConsumerApi,
        IUserTaskConsumerApi
    { }
}
